packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -201,3 +648,843 @@ export const InputBoxButton = (props: InputBoxButtonProps): JSX.Element => {
+          result +=
+            htmlText.substring(lastProcessedHTMLIndex, tag.openTagIdx + tag.openTagBody.length + startChangeDiff) +
+            processedChange;
+
+          processedChange = '';
+          lastProcessedHTMLIndex = tag.openTagIdx + tag.openTagBody.length + endChangeDiff;
+          // the change is handled; exit
+          break;
+        }
+      } else if (startIndex > tag.plainTextBeginIndex && oldPlainTextEndIndex > plainTextEndIndex) {
+        console.log('updateHTML 1.3 result', result);
+        //the change started in the tag but finishes somewhere further
+        const startChangeDiff = startIndex - tag.plainTextBeginIndex - mentionTagLength;
+        if (isMentionTag) {
+          const [resultValue, updatedChange, htmlIndex] = handleMentionTagUpdate(
+            htmlText,
+            oldPlainText,
+            lastProcessedHTMLIndex,
+            oldPlainText.substring(startIndex, startIndex + 1) !== ' ' && change === '' ? ' ' : '', // if substring !== ' ' && change is empty -> the change should be " " and not empty string but " " wasn't included in change; otherwise the part of mention should be just deleted without processedChange update
+            tag,
+            closeTagIdx,
+            closeTagLength,
+            plainTextEndIndex,
+            startIndex,
+            oldPlainTextEndIndex,
+            mentionTagLength
+          );
+          result += resultValue;
+          lastProcessedHTMLIndex = htmlIndex;
+          processedChange = updatedChange;
+          // no need to handle plainTextSelectionEndIndex as the change will be added later
+          //proceed with the next calculations
+        } else if (tag.subTags !== undefined && tag.subTags.length !== 0 && tag.content !== undefined) {
+          // with subtags
+
+          // before the tag content
+          const stringBefore = htmlText.substring(lastProcessedHTMLIndex, tag.openTagIdx + tag.openTagBody.length);
+          lastProcessedHTMLIndex = closeTagIdx;
+
+          const [content] = updateHTML(
+            tag.content,
+            oldPlainText,
+            newPlainText,
+            tag.subTags,
+            startIndex - mentionTagLength,
+            oldPlainTextEndIndex - mentionTagLength,
+            '', // the part of the tag should be just deleted without processedChange update and change will be added after this tag
+            mentionTrigger
+          );
+          result += stringBefore + content;
+          //proceed with the next calculations
+        } else {
+          // no subtags
+          result += htmlText.substring(
+            lastProcessedHTMLIndex,
+            tag.openTagIdx + tag.openTagBody.length + startChangeDiff
+          );
+          lastProcessedHTMLIndex = closeTagIdx;
+          //proceed with the next calculations
+        }
+      } else if (startIndex < tag.plainTextBeginIndex && oldPlainTextEndIndex > plainTextEndIndex) {
+        console.log('updateHTML 1.4 result', result);
```

Comment:
I guess we can just search for all console.log then remove them, I will just ignore the rest of them in the PR

Reply:
All of them will be removed in the next PRs for this feature as this is WIP right now

Reply:
Removed